docs/public/s2i-to-devfile.adoc

```diff
@@ -0,0 +1,92 @@
+== Migrating from S2I components to Devfile components
+
+Devfiles are new way of deploying component with odo that offers more convenience and flexibility. Read more about devfiles https://devfile.github.io/[here]
+
+Before 2.0 release odo deploys component with a different mechanism called as s2i based components. The newer way of deploying components is with devfiles.
+
+Exisiting users using s2i based components could migrate to devfile component by following the below steps:
+
+. Change the current directory to the component directory you want to migrate.
++
+[source,sh]
+----
+ $ cd <directory-name>
+----
+
+. Check the component status 
++
+[source,sh]
+----
+ $ odo list
+
+Openshift Components: 
+APP     NAME      PROJECT        TYPE       SOURCETYPE     STATE
+app     myapp     default        nodejs     local          Pushed
+
+----
+
+. Convert s2i component to devfile component, it would convert the s2i based configuration files to devfile configuration files.
++
+[source,sh]
+----
+ $ odo utils convert-to-devfile
+ 
+ devfile.yaml is available in the current directory.
+
+----
+
+. Push the devfile component to the cluster
++
+[source,sh]
+----
+ $ odo push 
+ 
+ Validation
+  ✓  Validating the devfile [40136ns]
+
+ Creating Kubernetes resources for component myapp
+  ✓  Waiting for component to start [12s]
+
+ Applying URL changes
+  ✓  URL myapp-8080: http://myapp-8080-myapp-default.apps-crc.testing/ created
```

Comment:
we want to avoid all examples that show `crc`. Instead, use `example.com`. This is used on the official kubernetes docs as well as our documentation here: https://odo.dev/docs/deploying-a-devfile-using-odo/